src/software/geom/vector.cpp

```diff
@@ -99,6 +99,11 @@ Angle Vector::orientation() const
     return Angle::fromRadians(std::atan2(y_, x_));
 }
 
+double Vector::angleBetween(Vector &other) const
+{
+    return std::acos(dot(other) / (length() * other.length()));
+}
```

Comment:
do you need this function? could you do `vector.orientation().minDiff(other.orientation())`?